test/unit/preprocessing/linear_dep_remover/test_sklearn_compat.py

```diff
@@ -7,10 +7,31 @@
 import numpy as np
 import pandas as pd
 import pytest
-from sklearn.utils.estimator_checks import parametrize_with_checks
+from sklearn.utils.estimator_checks import check_estimator, parametrize_with_checks
 
 from fairlearn.preprocessing import CorrelationRemover
 
+EXPECTED_FAILED_CHECKS = {
+    "CorrelationRemover": {
+        "check_transformer_data_not_an_array": (
+            "this estimator only accepts pandas dataframes or numpy ndarray as input."
+        ),
+        "check_estimators_empty_data_messages": (
+            "this estimator raises on missing sensitive features in data.",
+        ),
+    },
+}
+
+
+def check_extended(estimator, check, expected_failed_checks=None):
+    try:
+        check_estimator(
+            estimator, expected_failed_checks=EXPECTED_FAILED_CHECKS[estimator.__class__.__name__]
+        )
+    except TypeError:
+        # sklearn version < 1.6
+        check(estimator)
```

Comment:
`check_estimator` runs the whole common test suite, not only a single test. We need to have a fix for `parametrize_with_checks` and not `check_estimator`

Reply:
just for transparency if anyone else peaks here: I misunderstood @adrinjalali when we discussed this change, it needs a bigger implementation. they will work on it in `sklearn_compat` and then we can take an example and implement it for our use case.